spec/features/admin_node_update_feature_spec.rb

```diff
@@ -41,20 +41,30 @@
     end
     # rubocop:enable RSpec/MultipleExpectations
 
+    # rubocop:disable RSpec/ExampleLength
     scenario "User clicks on 'Reboot to update'", js: true do
-      allow(::Velum::Salt).to receive(:call)
+      allow(::Velum::Salt).to receive(:call).and_return(true)
 
       # clicks on "Update admin node"
       find(".update-admin-btn").click
 
+      # wait modal to appear
+      wait_until { page.has_text?("Reboot to update") }
+
       # clicks on "Reboot to update"
       find(".reboot-update-btn").click
 
-      # rubocop:disable RSpec/MessageSpies
-      expect(::Velum::Salt).to receive(:call)
-      # rubocop:enable RSpec/MessageSpies
+      wait_until { page.has_text?("Rebooting...") }
```

Comment:
It retries the expectation/find until they pass for up to the wait time. If the wait time expires the expectation/find fails and raises an error. You could set longer but any reasonable UI shouldn’t be taking more than 10 seconds max in a test environment, and usually less